hs_core/templates/pages/genericresource.html

```diff
@@ -2537,6 +2540,53 @@ <h3 style="color: red" id="file-type-error"></h3>
                 return false;
             }
 
+            // Enables and disables granting access buttons accordingly to the current access level
+            function updateActionsState(privilege){
+                // Set the state of dropdown menu items and remove buttons to false by default
+                $("form[data-access-type]").parent().addClass("disabled");
+                $("#list-roles a[data-role]").parent().addClass("disabled");
+                $(".access-table li.active[data-access-type]").closest("tr").addClass("hide-actions");
+
+                if (privilege == "view") {
+                    // Dropdown menu items
+                    $("form[data-access-type='Can view']").parent().removeClass("disabled");
+                    $("#list-roles a[data-role='view']").parent().removeClass("disabled");
+
+                    // Remove buttons
+                    $(".access-table li.active[data-access-type='Can view']").closest("tr").removeClass("hide-actions");
+                }
+                else if(privilege == "change") {
+                    // Dropdown menu items
+                    $("form[data-access-type='Can view']").parent().removeClass("disabled");
+                    $("#list-roles a[data-role='view']").parent().removeClass("disabled");
+
+                    $("form[data-access-type='Can edit']").parent().removeClass("disabled");
+                    $("#list-roles a[data-role='edit']").parent().removeClass("disabled");
+
+                    // Remove buttons
+                    $(".access-table li.active[data-access-type='Can view']").closest("tr").removeClass("hide-actions");
+                    $(".access-table li.active[data-access-type='Can edit']").closest("tr").removeClass("hide-actions");
+                }
+                else if(privilege == "owner") {
+                    // Dropdown menu items
+                    $("form[data-access-type='Can view']").parent().removeClass("disabled");
+                    $("#list-roles a[data-role='view']").parent().removeClass("disabled");
+
+                    $("form[data-access-type='Can edit']").parent().removeClass("disabled");
+                    $("#list-roles a[data-role='edit']").parent().removeClass("disabled");
+
+                    $("form[data-access-type='Is owner']").parent().removeClass("disabled");
+                    $("#list-roles a[data-role='owner']").parent().removeClass("disabled");
+
+                    // Remove buttons
+                    $(".access-table li.active[data-access-type='Can view']").closest("tr").removeClass("hide-actions");
+                    $(".access-table li.active[data-access-type='Can edit']").closest("tr").removeClass("hide-actions");
+                    if ($(".access-table li.active[data-access-type='Is owner']").length > 1) {     // At least one owner constrain
```

Comment:
do we allow owners to remove other owners when there are multiple owners or we only allow owners to remove themselves? If the former, this logic works. If the latter, more check may be needed to make sure owner A cannot remove owner B.